### PR TITLE
Add Use Cases to DG

### PR DESCRIPTION
**Pull request by colinhia**
Add Preliminary Use Cases to DG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -325,11 +325,62 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use case resumes at step 1.
+
+**Use case: U4. Add remarks about a person**
+
+**MSS**
+
+1.  User performs <u>View a person (U3)</u> 
+2.  AgentAssist returns the person
+3.  User requests to add a remark about the person
+4.  AgentAssist adds a comment for the person
```
Comment (ZShunRen) [inline docs/DeveloperGuide.md:372]: To keep it consistent, perhaps replace 'comment' with 'remark'

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -304,16 +304,16 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `AgentAssist` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: U1. Delete a person**
```
Comment (ZShunRen) [inline docs/DeveloperGuide.md:309]: Perhaps you can replace `U1.` with `U1 - ...`

**Review by ZShunRen — CHANGES_REQUESTED** [review]
Overall, great effort Colin! I just have a few nitpicks about the style.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (colinhia) [reply docs/DeveloperGuide.md]: Thanks for the feedback! Amended in the following commit

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (colinhia) [reply docs/DeveloperGuide.md]: Thanks for the catch! Amended in the following commit

**Review by FionaQY — APPROVED** [review]
LGTM

**Review by ZShunRen — APPROVED** [review]
LGTM!

**Pull request merged**